### ISSUE TITLE
Missing ResourceWatcherDelegate declaration

### ISSUE THREAD

**Issue by segabor** [swiftkube/servicediscovery#2]
`ServiceDiscoveryDelegate` wants to implement a now abandoned protocol. I guess it requires a redesign.

**Comment by iabudiab** [comment]
@segabor exactly, as I've mentioned in the first issue, an overhaul was started but it is still a WIP.

Thank you again for raising the issues here. Even the slightest attention theses packages received, give me a boost to dedicate more time and to not neglect them.👍

**Comment by iabudiab** [comment]
Covered by #1
